src/parse.c

```diff
@@ -8424,6 +8424,9 @@ static void p_waveform(tree_t stmt, type_t constraint)
 
    while (optional(tCOMMA))
       tree_add_waveform(stmt, p_waveform_element(constraint));
+
+   if (set_loc)
```

Comment:
Can't this be done unconditionally? The only other place this is called is from the sequential signal assignment statement, which will overwrite the location anyway.

Reply:
True, I have changed it.